src/parquet/column/properties.h

```diff
@@ -128,16 +129,25 @@ class WriterProperties {
       return this;
     }
 
-    Builder* set_compression(const std::shared_ptr<schema::ColumnPath>& path,
-                             Compression::type codec) {
-      codecs_[path->ToDotString()] = codec;
+    Builder* compression(Compression::type codec) {
+      default_codec_ = codec;
       return this;
     }
 
+    Builder* compression(const std::string& path, Compression::type codec) {
+      codecs_[path] = codec;
+      return this;
+    }
+
+    Builder* compression(const std::shared_ptr<schema::ColumnPath>& path,
+                         Compression::type codec) {
+      return this->compression(path->ToDotString(), codec);
+    }
+
     std::shared_ptr<WriterProperties> build() {
       return std::shared_ptr<WriterProperties>(new WriterProperties(
           allocator_, dictionary_enabled_, dictionary_pagesize_,
-          pagesize_, version_, codecs_));
+          pagesize_, version_, default_codec_, std::move(codecs_)));
```

Comment:
If we move, the builder can only be used once. We probably want to have it reusable. 

@wesm @majetideepak got an opinion on that?